diff --git a/dist/tres.js b/dist/tres.js
--- a/dist/tres.js
+++ b/dist/tres.js
@@ -792,12 +792,14 @@
 		fpsLimit
 	});
 	const start = () => {
+		if (isActive.value) return;
 		timer.start();
 		resume();
 	};
 	const stop = () => {
+		if (!isActive.value) return;
+		pause();
 		timer.stop();
-		pause();
 	};
 	return {
 		start,
@@ -967,8 +969,9 @@
 		renderer.shadowMap.type = options.shadowMapType ?? fallback;
 		forceMaterialUpdate();
 	});
-	onUnmounted(() => {
-		renderer.dispose();
+	onUnmounted(async () => {
+		loop.stop();
+		await renderer.dispose();
 		if ("forceContextLoss" in renderer) renderer.forceContextLoss();
 	});
 	return {
