diff --git a/dist/tres.js b/dist/tres.js
index ca514200cc5adc4889d19b56ca2fa7c193ac6801..fb6bfdbcdecc3e04e983482f71a25934850d7ef1 100644
--- a/dist/tres.js
+++ b/dist/tres.js
@@ -791,12 +791,14 @@ const useCreateRafLoop = (cycleFn, { fpsLimit } = {}) => {
 		fpsLimit
 	});
 	const start = () => {
+		if (isActive.value) return;
 		timer.start();
 		resume();
 	};
 	const stop = () => {
-		timer.stop();
+		if (!isActive.value) return;
 		pause();
+		timer.stop();
 	};
 	return {
 		start,
@@ -970,6 +972,7 @@ Error details: ${rendererError.message}`, rendererError);
 		forceMaterialUpdate();
 	});
 	onUnmounted(() => {
+		loop.stop();
 		renderer.dispose();
 		if ("forceContextLoss" in renderer) renderer.forceContextLoss();
 	});
